sympy/utilities/tests/test_codegen.py

```diff
@@ -508,6 +508,46 @@ def test_output_arg_c_reserved_words():
     assert result[0][1] == expected
 
 
+def test_multidim_c_argument_cse():
+    A_sym = MatrixSymbol('A', 3, 3)
+    b_sym = MatrixSymbol('b', 3, 1)
+    A = Matrix(A_sym)
+    b = Matrix(b_sym)
+    c = A*b
+    cgen = CCodeGen(project="test", cse=True)
```

Comment:
I had to use cse so that the code reached the section where c locals are printed. If I did not use cse, then there where no locals and the "raise CodeGenError("Only column vectors are supported in local variabels. Local result {} has dimensions {}".format(result, dims))" line was not reached. 

The test is a bit simple as it does not show cse being used but it does however force an array from a multi-dimensional input.

Reply:
of course! Thanks.